cwms-data-api/src/main/java/cwms/cda/security/OpenIDAccessManager.java

```diff
@@ -68,48 +66,22 @@ public OpenIDAccessManager(String wellKnownUrl, String issuer, int realmKeyTimeo
 
     @Override
     public void manage(Handler handler, Context ctx, Set<RouteRole> routeRoles) throws Exception {
-        dataSource = ctx.attribute(ApiServlet.DATA_SOURCE);
-        String user = getUserFromToken(ctx);
-        prepareContextWithUser(ctx, user);
+        Optional<DataApiPrincipal> p = getUserFromToken(ctx);
+        AuthDao.isAuthorized(ctx,p,routeRoles);
+        AuthDao.prepareContextWithUser(ctx, p.get());
         handler.handle(ctx);
     }
 
 
-    /**
-     * Allows connection to be correctly setup. User name already verified
-     * by signed JWT. Just assert to system.
-     *
-     * @param ctx javalin context if additional parameters are required.
-     * @param user username, which is ignored except a log message
-     */
-    private void prepareContextWithUser(Context ctx, String user) throws SQLException {
-
-
-        ConnectionPreparer userPreparer = new DirectUserPreparer(user);
-        ConnectionPreparer officePrepare = new SessionOfficePreparer(ctx.queryParam(Controllers.OFFICE));
-        DelegatingConnectionPreparer apiPreparer =
-            new DelegatingConnectionPreparer(officePrepare,userPreparer);
 
-        if (dataSource instanceof ConnectionPreparingDataSource) {
-            ConnectionPreparingDataSource cpDs = (ConnectionPreparingDataSource)dataSource;
-            ConnectionPreparer existingPreparer = cpDs.getPreparer();
-
-            // Have it do our extra step last.
-            cpDs.setPreparer(new DelegatingConnectionPreparer(existingPreparer, apiPreparer));
-        } else {
-            ctx.attribute(ApiServlet.DATA_SOURCE,
-                          new ConnectionPreparingDataSource(apiPreparer, dataSource));
-        }
-    }
-
-    private String getUserFromToken(Context ctx) {
+    private Optional<DataApiPrincipal> getUserFromToken(Context ctx) throws CwmsAuthException {
```

Comment:
return is never an empty Optional. Should just return DataApiPrincipal instead

Reply:
Yeah, I mostly just did that to copy the structure of others. I plan to rearrange all this later to pass the generated DataApiPrincipal to a single AccessManager so all of this should be considered a stopgap.

Reply:
Sounds good, I got worried about the unchecked `Optional.get()` but then saw there was no nullable `Optional`